src/sentry/integrations/services/repository/impl.py

```diff
@@ -89,6 +89,29 @@ def update_repository(self, *, organization_id: int, update: RpcRepository) -> N
 
             repository.save()
 
+    def update_repositories(self, *, organization_id: int, updates: list[RpcRepository]) -> None:
+        with transaction.atomic(router.db_for_write(Repository)):
+            update_mapping: dict[int, dict[str, Any]] = {}
+
+            updated_fields: set[str] = set()
+            for update in updates:
+                update_dict = update.dict()
+                update_dict.pop("id", None)  # Remove id field
+                update_mapping[update.id] = update_dict
+                updated_fields.update(update_dict.keys())
+
+            repositories = Repository.objects.filter(
+                organization_id=organization_id, id__in=update_mapping.keys()
+            )
+
+            # Apply updates to each repository object
+            for repository in repositories:
+                if repo_update := update_mapping.get(repository.id):
```

Comment:
This check seems redundant. Can't we guarantee a mapping from the previous step? Letting this raise on a key error may be preferred if it means flagging logic issues.